controllers/llamaCPP.cc

```diff
@@ -681,62 +705,47 @@ bool llamaCPP::LoadModelImpl(std::shared_ptr<Json::Value> jsonBody) {
     params.model_alias = params.model;
   }
 
-  llama_backend_init();
-
-  // LOG_INFO_LLAMA("build info",
-  //                {{"build", BUILD_NUMBER}, {"commit", BUILD_COMMIT}});
-  LOG_INFO_LLAMA("system info",
-                 {
-                     {"n_threads", params.n_threads},
-                     {"total_threads", std::thread::hardware_concurrency()},
-                     {"system_info", llama_print_system_info()},
-                 });
+  if (ShouldInitBackend()) {
+    llama_backend_init();
+    LOG_INFO_LLAMA("system info",
+                   {
+                       {"n_threads", params.n_threads},
+                       {"total_threads", std::thread::hardware_concurrency()},
+                       {"system_info", llama_print_system_info()},
+                   });
+  }
 
+  auto model_id = nitro_utils::getModelId(*jsonBody);
   // load the model
-  if (!llama.load_model(params)) {
+  if (!server_ctx_map[model_id].load_model(params)) {
     LOG_ERROR << "Error loading the model";
     return false;  // Indicate failure
   }
-  llama.initialize();
-
-  if (queue != nullptr) {
-    delete queue;
+  server_ctx_map[model_id].initialize();
+
+  if (inference_task_queue == nullptr ||
+      task_queue_thread_num < params.n_parallel) {
+    task_queue_thread_num = std::max(task_queue_thread_num, params.n_parallel);
+    LOG_INFO << "Start inference task queue, num threads: "
+             << task_queue_thread_num;
+    inference_task_queue = std::make_unique<trantor::ConcurrentTaskQueue>(
```

Comment:
The inference task queue is constructed with the model's parameters. It is shared among the model's operations, which is incorrect, right? Let's say it would be renewed every time a new model request with a larger n_parallel number comes in:
1. How will pending tasks be processed since it is renewed?
2. Will later requests change thread_num, causing side effects for previously loaded models?"

Reply:
One solution (though it is not optimal) is that we can use a task queue for each model. Will change the code to make it works correctly first.